src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Layers/FeatureLayerDefinitionExpression/FeatureLayerDefinitionExpression.xaml.cs

```diff
@@ -1,79 +1,147 @@
-            MyMapView.Map = myMap;
+        private void Expression_Click(object sender, RoutedEventArgs e)
+        {
+            // Reset the display filter definition.
+            _featureLayer.DisplayFilterDefinition = null;
 
-            // Create the uri for the feature service
-            Uri featureServiceUri = new Uri(
-                "https://sampleserver6.arcgisonline.com/arcgis/rest/services/SF311/FeatureServer/0");
+            // Set the definition expression to show specific features only.
+            _featureLayer.DefinitionExpression = "req_Type = 'Tree Maintenance or Damage'";
+
+            CountFeatures();
+        }
 
-            // Initialize feature table using a url to feature server url
-            ServiceFeatureTable featureTable = new ServiceFeatureTable(featureServiceUri);
+        private void Filter_Click(object sender, RoutedEventArgs e)
+        {
+            // Disable the feature layer definition expression
+            _featureLayer.DefinitionExpression = "";
 
-            // Initialize a new feature layer based on the feature table
-            _featureLayer = new FeatureLayer(featureTable);
+            // Set the display filter definition on the layer.
+            _featureLayer.DisplayFilterDefinition = _definition;
 
-            // Add the feature layer to the map
-            myMap.OperationalLayers.Add(_featureLayer);
-  
+            CountFeatures();
         }
 
-        private void OnApplyExpressionClicked(object sender, RoutedEventArgs e)
+        private void Reset_Click(object sender, RoutedEventArgs e)
         {
-            // Adding definition expression to show specific features only
-            _featureLayer.DefinitionExpression = "req_Type = 'Tree Maintenance or Damage'";
+            // Disable the feature layer definition expression
+            _featureLayer.DefinitionExpression = "";
+
+            // Reset the display filter definition.
+            _featureLayer.DisplayFilterDefinition = null;
+
+            CountFeatures();
         }
 
-        private void OnResetButtonClicked(object sender, RoutedEventArgs e)
+        private void OnViewpointChanged(object sender, EventArgs e)
         {
-            // Reset the definition expression to see all features again
-            _featureLayer.DefinitionExpression = "";
+            CountFeatures();
+        }
+
+        private void CountFeatures()
+        {
+            // Use a DispatcherTimer to prevent concurrent feature counts.
+            try
+            {
+                if (_timer == null || !_timer.IsEnabled)
+                {
+                    if (_timer == null)
+                    {
+                        _timer = new DispatcherTimer
+                        {
+                            Interval = TimeSpan.FromSeconds(1),
+                        };
+                        _timer.Tick += (a, b) =>
+                        {
+                            _ = QueryFeatureCountAsync();
```

Comment:
I'd like to see a Cancellation-based workflow here, rather than the timer-based approach, or the stored task+bool to retry approach used on other platforms.